index.js

```diff
@@ -78,9 +82,23 @@ AudioManager.prototype.init = function () {
 };
 
 //▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
-/** Get a unused sound object (or a new one if no more empty sounds are available in pool) */
-AudioManager.prototype.getEmptySound = function () {
+/** Get a unused sound object (or a new one if no more empty sounds are available in pool)
+ *
+ * @param {string} channelId - channel id where the sound will be used
+ * @param {string} soundId   - sound id
+ *
+ * @return {Object} sound object
+ */
+AudioManager.prototype.getEmptySound = function (channelId, soundId) {
 	var sound;
+
+	// custom sound constructor
+	var constructor = this.settings.getSoundConstructor(channelId, id);
+	if (constructor) {
+		sound = new constructor();
```

Comment:
That seems very unsafe. What if AudioManager is multiple levels up the prototype chain. I would rather you not mess with my class. If you just wanna check if the prototype is there at all, there are functions and operators for that (eg: `sound instanceof AudioManager`).